### ISSUE TITLE
Permissions issue with mint installation with mint

### ISSUE THREAD

**Issue by jonasman** [yonaskolb/Mint#188]
```
% swift run mint install yonaskolb/mint
🌱 Finding latest version of mint
🌱 Cloning mint 0.14.2
🌱 Resolving package
🌱 Building package
Error: You don’t have permission to save the file “0.14.2” in the folder “build”. 
```

i solved by installing with:
`% sudo swift run mint install yonaskolb/mint`

**Comment by vknabel** [comment]
Hi @jonasman,

Using `sudo` to run mint can be a security risk, as it would execute tools with root permissions.
Probably there is at least one file or folder, which your current user cannot access.

Instead you can try to change the owner of your mint files using `sudo chown -R $(whoami) /usr/local/lib/mint/*`. 
Alternatively you could try to delete all data of mint located at `/usr/local/lib/mint`. 
Does it help you?

**Comment by ethanhuang13** [comment]
I have the same issue. However there is no `/usr/local/lib/mint` folder on my new Mac. Therefore I resolved the issue by creating this folder.

**Comment by ratkins** [comment]
I just had the same problem on a fresh install of macOS 11.1:

```
ratkins@girder ios-app % mint run swiftlint --config ".swiftlint.yml"
🌱 Cloning SwiftLint 0.41.0
🌱 Resolving package
🌱 Building package
Error: You don’t have permission to save the file “0.41.0” in the folder “build”.
```

I had to create `sudo mkdir -p /usr/local/lib/mint` and `sudo chown -R $(whoami):admin /usr/local/lib` to get it to work.

Is there a way to make this work out of the box?

**Comment by yonaskolb** [comment]
This is an issue with newer versions of macOS.
Without using sudo, we could move installations from `/usr/local/lib/mint` to `~/.mint`, though this is a breaking change. But that could maybe be mitigated by reading from both directories.

**Comment by ratkins** [comment]
How does Homebrew do it? I thought `/usr/local` on macOS was explicitly for the use of third party tools like brew and mint (I found a good reference for this once but my Google-fu is failing me.)

**Comment by guillaumealgis** [comment]
Same here.

It seems like for version of macOS before Big Sur, `/usr/local/bin` was 

```
rwxrwxr-x yourusername admin 
```

and on Big Sur its

```
rwxr-xr-x root wheel
```

I'm not sure whether its a macOS change (which would make sense since `/usr/local` is shared among users of the machine), or if Homebrew was setting different ownership / permission on `/usr/local/bin`)

-----

Anyway, for the time being setting custom `MINT_PATH` and `MINT_LINK_PATH` in my shell environment solved the problem for me.

I used:

```bash
export MINT_PATH="$HOME/.mint"
export MINT_LINK_PATH="$MINT_PATH/bin"
```

and added `$MINT_LINK_PATH` to my `$PATH`.

----------------

@ratkins where you referring to this? [https://support.apple.com/en-us/HT204899](https://support.apple.com/en-us/HT204899)

> Paths and apps that third-party apps and installers can continue to write to include:
> - /Applications
> - /Library
> - /usr/local

**Comment by Jeehut** [comment]
@guillaumealgis Thank you very much, your suggestion worked like a charm! 🎉 

But I adjusted the commands so the new env vars are persisted:

```bash
echo 'export MINT_PATH="$HOME/.mint"' >> ~/.zshrc
echo 'export MINT_LINK_PATH="$MINT_PATH/bin"' >> ~/.zshrc
source ~/.zshrc
```

**Comment by aonez** [comment]
@yonaskolb could probably mint check it it needs to use `/opt/ (as @ratkins suggests) instead of `/usr/local/' in Big Sur?

**Comment by yonaskolb** [comment]
If updated the path defaults in this PR https://github.com/yonaskolb/Mint/pull/216. Can people have a look and give feedback. One outcome is that `~/.mint/bin` must now be manually added to $PATH if packages need to be run outside of mint

**Comment by yonaskolb** [comment]
Released in 0.17.0

**Comment by ismael-gobernado-deltatre** [comment]
The same issue is happening in Ventura with an M1, Fix is valid
